doc/developers/contributing.rst

```diff
@@ -603,6 +603,37 @@ Finally, follow the formatting rules below to make it consistently good:
     SelectKBest : Select features based on the k highest scores.
     SelectFpr : Select features based on a false positive rate test.
 
+* When documenting the parameters and attributes, here is a list of some good
+  examples::
+
+    some_param : {'hello', 'goodbye'}, bool or int, default=True
+	    The parameter description goes here, which can be either a string
+        literal (either `hello` or `goodbye`), a bool, or an int. The default
+        value is True.
+
+    array_parameter : {array-like, sparse matrix, dataframe} of shape (n_samples, n_features) or (n_samples,)
+        This parameter accepts a data in either of the mentioned forms, with one
+        of the mentioned shapes.
+
+In general have the following in mind:
+
+    1. Do not use optional. Use `default=`.
+    2. Python basic types. (`bool` instead of `boolean`)
+    3. When defining 1-D shape, use parenthesis. `array-like of shape (n_samples,), None, default=None`.
+    4. When defining 2-D shape, use parenthesis. `array-like of shape (n_samples, n_features)`.
+    5. str with multiple options: `input: {'log', 'squared', 'multinomial'}`
+    6. Only use `or` for separating types. `float, int or None, default=None`
```

Comment:
I'd remove 6 and 7 since we don't even follow these in the examples above.

We use or for separating shapes, and we use commas everywhere in the examples, not just for separating types.

Reply:
I agree with remove 6 and 7.